### PR TITLE
Set retry time to 60 seconds for auth timeouts.

### PR DESCRIPTION
**Pull request by ghmj2417**
This allows for up to 60 seconds of trying to read data from the device during the initial connection.  This is in relation to issue #257  where TACACS, RADIUS, etc are not available and the device may not respond in a timely fashion.  I tested this in my environment and I was able to connect to my devices that have RADIUS as primary and the servers were unavailable.

Couple things: changed the max loop to 60 to resemble 60 seconds.  Changed i += 1 to using main_delay, reason for this; we should probably wait a consistent amount of time and not rely on someone setting global_delay_factor. Using main_delay mimics actual wait time. With this change only affecting the establishment of the connection, I feel 60 seconds is ample amount of time to wait for authentication.  I don't know if I feel allowing a user to specify a high amount for global_delay_factor is the best idea for fixing the timeout issue especially since a global delay factor affects other items.  I know it's best to try and keep things dynamic but I have no problem with this setting being static.

Thoughts?

### CODE REVIEW

**Comment by ghmj2417** [comment]
If you want the user to be able to customize the retry time, I can do that but we will need to add another arg into __init__, maybe call it auth_timeout?  Maybe set it to 20 by default?

**Comment by ktbyers** [comment]
Increasing the count to 60 is fine.

Why do this?

                i += main_delay

I didn't understand why you wanted to do that. main_delay will start at .1 and then increase up to a maximum value of 8.

I would propose increase the:

while i <= 60:

but just do the:

                 i += 1

Does this fix your issue?

global_delay_factor will still affect it because of this:

delay_factor = self.select_delay_factor(delay_factor=0)

Basically, I want all sleeps to be affected by global_delay_factor so that ultimately the end-user can adjust this if necessary.

**Comment by ktbyers** [comment]
I think the solution is just to increase the loop count (while i <= 60). It will break out whenever it encounters the first data so devices that respond faster won't be slowed down.

**Comment by ghmj2417** [comment]
If you only change the loop to i <= 60 then at a minimum you could wait a total of 367.3228 seconds before you get to the exception (assuming no data is being read).  On top of the wait time, specifying 60 doesn't really give an indicator of how long you are actually waiting until the exception is thrown.

If you set i += main_delay then your i <= 60 is based off of actual wait time.  Example of setting i += main_delay, let's say main_delay is set to 3, on your first iteration through the loop you will be waiting 3.3 seconds, i now reflects the actual wait time.  Your next iteration, main_delay will equal 3.63, however i will equal 6.93 which reflects actual waiting time.  Hence, now i <= 60 reflects time in seconds of waiting.  I personally like this approach better and that's why I suggested I could add an auth_timeout variable so we can make it i <= self.auth_timeout, instead of i <= 60.  This gives the user flexibility if their environment takes awhile for authentications.

**Comment by ktbyers** [comment]
@ghmj2417 How long a wait does TACACS+/RADIUS require in your case?

There is already an argument for auth_timeout in __init__ the timeout argument. This is actually a post authentication timeout.

We could possibly base the timeout off that. 

I don't think we should spend too much time here. The key thing is just making sure TACACS+ has enough time to clear with reasonable margin to account for different environments. We should give end-users a way to adjust the time if they require using global_delay_factor

**Comment by ghmj2417** [comment]
My scenario, I need at ~8 seconds but I setup my authentication servers for only 2 retries and a timeout of 2.  I think the default setting on a Cisco device is 3 retries with a 5 second timeout.

Last commit should work.  I increased the loop to 40 and changed i back to i += 1.  This, by default, should allow approximately 53 seconds to read data.  With this configuration, users can extend the retry timeout by setting the global_delay_factor.
